### PR TITLE
explanatory error on `--print target-spec-json` without unstable options

### LINKED ISSUE #41683: Print note when trying to use unstable --print target-spec-json
#38061 introduced `--print target-spec-json` but it's a little hard to find out how to use it:

Current:
```
$ rustc +nightly --print target-spec-json
error: unknown print request `target-spec-json`
```
Expected:
```
$ rustc +nightly --print target-spec-json
error: the `-Z unstable-options` flag must also be passed to enable the print request `target-spec-json`
```

### PR DESCRIPTION
**Pull request by zackmdavis**
unstable_target_spec

Resolves #41683.

### CODE REVIEW

File: src/librustc/session/config.rs [hunk 1]
```diff
@@ -1600,6 +1600,11 @@ pub fn build_session_options_and_crate_config(matches: &getopts::Matches)
             "code-models" => PrintRequest::CodeModels,
             "target-spec-json" if nightly_options::is_unstable_enabled(matches)
                 => PrintRequest::TargetSpec,
+            "target-spec-json" if !nightly_options::is_unstable_enabled(matches) => {
```
Comment (Mark-Simulacrum) [inline src/librustc/session/config.rs]: Could you unify the if statement here, making it an if-else statement?

**Comment by arielb1** [comment]
r+ modulo that nit

**Comment by zackmdavis** [comment]
(force-push addresses nit)

**Pull request merged**